bn_mp_to_radix.c

```diff
@@ -5,66 +5,118 @@
 
 /* stores a bignum as a ASCII string in a given radix (2..64)
  *
- * Stores upto maxlen-1 chars and always a NULL byte
+ * Stores upto "size - 1" chars and always a NULL byte, puts the number of characters
+ * written, including the '\0', in "written".
+ * For "size == 0u" the whole number in "a" will get converted.
  */
-mp_err mp_to_radix(const mp_int *a, char *str, size_t maxlen, int radix)
+mp_err mp_to_radix(const mp_int *a, char **str, size_t *length, int radix)
 {
-   int     digs;
+   size_t digs;
    mp_err  err;
    mp_int  t;
    mp_digit d;
-   char   *_s = str;
+   char   *_s;
+   size_t len;
 
-   /* check range of the maxlen, radix */
-   if ((maxlen < 2u) || (radix < 2) || (radix > 64)) {
+   /* If we want to fill a bucket we at need a bucket in the first place. */
+   if (str == NULL) {
+      return MP_VAL;
+   }
+
+   /* check range of radix and size*/
+   if ((radix < 2) || (radix > 64)) {
       return MP_VAL;
    }
 
    /* quick out if its zero */
    if (MP_IS_ZERO(a)) {
-      *str++ = '0';
-      *str = '\0';
+      *str = MP_MALLOC(2u);
+      if (*str == NULL) {
+         return MP_MEM;
+      }
+      _s = *str;
+      **str = '0';
+      (*str)++;
+      **str = '\0';
+      if (length != NULL) {
+         *length = 2u;
+      }
+      *str = _s;
```

Comment:
just work with `_s` above and you don't need to fixup `str` here

Reply:
Ah, my beloved leftovers! ;-)
Thanks!

No, I don't know what is was that left this, too many variations over the length of this thread.
I know, I really should stop trying to keep up with every single move \*sigh\*